src/blazar_tempest_plugin/auth/oidc_provider.py

```diff
@@ -0,0 +1,87 @@
+from keystoneauth1 import loading
+from keystoneauth1 import session
+from tempest.lib.auth import KeystoneV3AuthProvider
+from tempest.lib.auth import KeystoneV3Credentials
+from oslo_log import log as logging
+
+
+LOG = logging.getLogger(__name__)
+
+
+class KeystoneV3OidcCredentials(KeystoneV3Credentials):
+
+    _EXTRA_ATTRS = [
+        "protocol",
+        "identity_provider",
+        "client_id",
+        "client_secret",
+        "access_token_type",
+        "discovery_endpoint",
+    ]
+
+    ATTRIBUTES = KeystoneV3Credentials.ATTRIBUTES + _EXTRA_ATTRS
+
+
+class KeystoneV3OidcAuthProvider(KeystoneV3AuthProvider):
+
+    def _get_auth(self):
+        """Acquire a token through keystoneauth1 v3oidcpassword plugin."""
+
+        # If the credentials do not have any of the extra attributes, fall
+        # back to the default v3 password flow.
+        if not any(getattr(self.credentials, attr, None) is not None
+                   for attr in getattr(self.credentials, '_EXTRA_ATTRS', [])):
+            return super()._get_auth()
+
+        LOG.debug("Authenticating with v3oidcpassword flow to %s",
+                  self.auth_url)
```

Comment:
My main concern here is that we're getting implicit behavior from the fallback, but this goes along with "do we override v3, or specify something new"

I would argue that the "right way" is to specify a new type of identity version "v3oidc...", and then if we provide credentials that don't work with it, it should be an explicit failure instead of an implicit success.

All that said, I think this is fine for now, ant the other parts can be postponed for a refactor.